cypress-io/cypress
# chore: Default dev mode to production cloud rather than staging

- Closes #23020

### User facing changelog


n/a

### Additional details


This PR defaults to using the production instance of cloud when the app is running in development mode. Before, it was using the staging instance. You can still target staging by setting `CYPRESS_INTERNAL_ENV=staging`.

This will allow us to see real CI runs for specs that are in our cypress packages while we develop

### Steps to test


1. Run the app with this branch
2. Go to the Runs tab and click on a run
3. Verify that you are taken to https://dashboard.cypress.io/ rather than https://dashboard-staging.cypress.io/
4. Verify that you can see runs in the latest runs in the specs list under the latest runs column
5. Set `CYPRESS_INTERNAL_ENV=staging` and re-run the application
6. Repeat step 2 and verify that you are now taken to https://dashboard-staging.cypress.io/

### PR Tasks


- [na] Have tests been added/updated?
- [na] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
